linux-anvil-cuda/Dockerfile

```diff
@@ -11,15 +11,31 @@ ENV CUDA_VER ${CUDA_VER}
 # Set an encoding to make things work smoothly.
 ENV LANG en_US.UTF-8
 
-# Set path to CUDA install.
+# Set path to CUDA install (this is a symlink to /usr/local/cuda-${CUDA_VER})
 ENV CUDA_HOME /usr/local/cuda
+# Set additional LD_LIBRARY_PATH entry for path of libcuda.so
+ENV LD_LIBRARY_PATH ${LD_LIBRARY_PATH}:${CUDA_HOME}/compat-${CUDA_VER}
 
 # Add a timestamp for the build. Also, bust the cache.
 ADD http://worldclockapi.com/api/json/utc/now /opt/docker/etc/timestamp
 
 # Resolves a nasty NOKEY warning that appears when using yum.
 RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
 
+# upstream image nvidia/cuda:9.2-devel-centos6 does not contain libcuda.so; it should be
+# installed in ${CUDA_HOME}/compat-${CUDA_VER}, however cuda-compat-${CUDA_VER}.x86_64.rpm
+# is only packaged for 10.x; we abuse cuda-compat-10.0 for this, since the actual dependency
+# containing libcuda for 9.2 (xorg-x11-drv-nvidia-libs) pulls in a huge amount of dependencies
+RUN if [ ${CUDA_VER} == "9.2" ]; then \
+        curl -O https://developer.download.nvidia.com/compute/cuda/repos/rhel6/x86_64/cuda-repo-rhel6-9.2.148-1.x86_64.rpm && \
+        rpm --install cuda-repo-*.rpm && \
+        rm cuda-repo-*.rpm && \
+        yum install -y cuda-compat-10-0.x86_64 && \
+        # note that $CUDA_HOME is just a symlink - see above
+        mv /usr/local/cuda-10.0/* /usr/local/cuda-9.2 && \
+        rm -rf /usr/local/cuda-10.0; \
+    fi
```

Comment:
The block above should probably go into `conda-forge-ci-setup-feedstock` as we can't ship this in our docker image.

Reply:
You mean license-wise? Not that I claim to understand the licensing intricacies, but how can we repackage the `nvidia/cuda:${CUDA_VER}-devel-centos6`, but not another library? And how do things change by doing this in the ci-setup?

In any case, if you prefer me to do this within https://github.com/conda-forge/conda-forge-ci-setup-feedstock/blob/master/recipe/run_conda_forge_build_setup_linux, then I can.

However, from my experience with [`ENV`](https://docs.docker.com/engine/reference/builder/#env), I'm pretty sure that LD_LIBRARY_PATH will be persisted.

Reply:
> Not that I claim to understand the licensing intricacies, but how can we repackage the 

We aren't repackaging it. Docker works in layers and this layer is distributed by nvidia and we are only distributing the ones on top.

Reply:
OK. I'll try to add this in `conda-forge-ci-setup-feedstock` and leave a comment here (or something).

I still think we should adapt add the LD_LIBRARY_PATH, as otherwise every gpu-recipe will have to set it to test any code that needs `libcuda.so`

Reply:
ENV LD_LIBRARY_PATH like this will not affect anything. That's why I said we want to use ldconfig.

Reply:
> ENV LD_LIBRARY_PATH like this will not affect anything.

I'm not sure I understand what you mean. In docker, this works:
```
xxx@yyy:~$ sudo docker build -t add_env -f - . <<EOF
> FROM nvidia/cuda:9.2-devel-centos6
> ENV LD_LIBRARY_PATH=/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/compat-9.2
> CMD /bin/bash
> EOF
[...]
Successfully tagged add_env:latest
xxx@yyy:~$ sudo docker run --rm -it add_env bash
[root@d67bbed552d9 /]# echo $LD_LIBRARY_PATH
/usr/local/nvidia/lib:/usr/local/nvidia/lib64:/usr/local/cuda/compat-9.2
```

Reply:
conda-build would override this path AFAIK. Can you check?

Reply:
Just checked, it is not being overridden by conda build, it just has to be passed explicitly (cf. [here](https://docs.conda.io/projects/conda-build/en/latest/user-guide/environment-variables.html#env-vars)):
```
build:
  script_env:
   - LD_LIBRARY_PATH
```

I also tested this by adding this to a recipe, `set LD_LIBRARY_PATH=C:/hello/there`, creating `bld.bat` as:
```
echo %LD_LIBRARY_PATH%
```
and running `conda build recipe`. Output:
```
(%BUILD_PREFIX%) %SRC_DIR%>echo C:/hello/there
C:/hello/there
```

Reply:
Passing LD_LIBRARY_PATH is not done by conda-build, because it's dangerous to do so and might break conda installed executables. This is probably not an on the docker image, but recipes shouldn't add it to avoid breaking building recipes by users outside docker image.

Reply:
Yes, OK, I didn't adapt the example from the docs, but the idea would be to be able to use this for gpu-recipes as:
```
test:
  script_env:
   - LD_LIBRARY_PATH
```

Which would then pick up the `LD_LIBRARY_PATH` we set in the docker image, and hence find `libcuda.so`.

Reply:
I'm not suggesting to use this during the build phase, I just illustrated the test you wanted me to perform.

Reply:
If we go with `ldconfig` route, recipes don't need to be fixed like that.

Reply:
IMO, using `LD_LIBRARY_PATH` in build phase and test phase are equally bad.

Reply:
OK, I relent. ;-)

I went and had a look at how this should be done, and found out. The sanest solution seems to be:
```
echo "$CUDA_HOME/compat-$CUDA_VER" >> /etc/ld.so.conf.d/cuda-$CUDA_VER.conf
```

The latter file exists already and contains (e.g.) `/usr/local/cuda-9.2/targets/x86_64-linux/lib`. 

Now the question is: are you fine with me fixin gthe `ldconfig` here, and bringing in `compat-9.2` in ci-setup? In the ci-setup, I don't have `$CUDA_VER` at my disposal, and this would make things much easier.

Reply:
> are you fine with me fixin gthe ldconfig here, and bringing in compat-9.2 in ci-setup? 

Yes, that's fine. Thanks.

Reply:
Opened https://github.com/conda-forge/conda-forge-ci-setup-feedstock/pull/93.